Dockerfile

```diff
@@ -1,21 +1,45 @@
-FROM quay.io/centos/centos:stream8
+# build poetry
+FROM quay.io/centos/centos:stream8 as poetry
 
 RUN dnf -y module install python39 && dnf -y install python39 python39-pip
-RUN mkdir /app
-ADD https://raw.githubusercontent.com/arcalot/arcaflow-plugin-template-python/main/LICENSE /app
-ADD example_plugin.py /app
-ADD test_example_plugin.py /app
-ADD requirements.txt /app
+
 WORKDIR /app
 
-RUN pip3 install -r requirements.txt
+COPY poetry.lock /app/
+COPY pyproject.toml /app/
+
+RUN python3.9 -m pip install poetry \
+ && python3.9 -m poetry config virtualenvs.create false \
+ && python3.9 -m poetry install --without dev \
+ && python3.9 -m poetry export -f requirements.txt --output requirements.txt --without-hashes
+
 
-RUN mkdir /htmlcov
-RUN pip3 install coverage
-RUN python3 -m coverage run test_example_plugin.py
-RUN python3 -m coverage html -d /htmlcov --omit=/usr/local/*
+# coverage test
+FROM poetry as coverage
```

Comment:
It is not clear to me what this stage buys us because it appears that this `FROM` is leveraging the entire result of the previous stage.

Reply:
My thought was to separate the building of poetry and the use of coverage. It is more aesthetics I guess for organization.  It makes it more clear what files are copied over and used to build poetry, and what files are used for testing. I can combine them if that is not needed

Reply:
Ok will do